receiver/podmanreceiver/libpod_client.go

```diff
@@ -151,3 +119,30 @@ func (c *podmanClient) ping(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (c *libpodClient) events(ctx context.Context, options url.Values) (<-chan Event, <-chan error) {
+	events := make(chan Event)
+	errs := make(chan error)
+	go func() {
+		resp, err := c.request(ctx, "/events", options)
+		if err != nil {
+			errs <- err
+		}
+		dec := json.NewDecoder(resp.Body)
+		for {
+			var e Event
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				err := dec.Decode(&e)
```

Comment:
Decode gets blocked until there is some data to decode (or the connection is closed). If the provided data cannot be transformed into an Event type (e) an error will be returned. Does that help answer your question?

Reply:
c.requests is only called once every time "event" is called right? When the body is done being decided what happen?

How many sub-functions will this function start?

Reply:
In the best scenario the scraper will just call the `events` function one and only one go routine will be spawned. The `c.request(ctx, "/events", options)` makes a request to a stream of data and the for loop blocks in the decoder till some data is available. 

If any error is encountered while decoding the body, it will be send to the errs channel. I am thinking that we might need to cancel that go routine if an error is found in the Event Loop: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10421/files/1da7180a672dcf4b2732550f22abf1dbbff04514#diff-8b60a921ebdaa31d678ccc8553509d3de656f33d052d58a98ded5bd9e6e94447R130

Reply:
Thanks for the explanation